[Swift in WebKit] Work towards modularizing PAL (part 1)

rr-codes opened the pull request:
#### e5981ec4a87f8f1739f4d9a31d46495d590670eb
<pre>
[Swift in WebKit] Work towards modularizing PAL (part 1)
<a href="https://bugs.webkit.org/show_bug.cgi?id=303206">https://bugs.webkit.org/show_bug.cgi?id=303206</a>
<a href="https://rdar.apple.com/165493410">rdar://165493410</a>

Reviewed by Mike Wyrzykowski.

Remove some settings from PAL and WebCore xcconfigs which are no longer used and/or no longer applicable

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:

Canonical link: <a href="https://commits.webkit.org/303763@main">https://commits.webkit.org/303763@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/33f2aae7a062dc63d5c3fe37d39f42cb32964b3e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/133547 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/6052 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/44705 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/141107 "Built successfully") | [  ~~🛠 win~~](https://ews-build.webkit.org/#/builders/59/builds/85599 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | ⏳ 🛠 ios-apple 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/135417 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/6572 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/5916 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/102165 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 win-tests~~](https://ews-build.webkit.org/#/builders/59/builds/85599 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/5326e87f-ada1-4111-a0e4-f55ef8eabecd) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/136494 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/162/builds/4660 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/119734 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/82961 "Passed tests") | | [![loading](https://user-images.githubusercontent.com/3098702/171232313-daa606f1-8fd6-4b0f-a20b-2cb93c43d19b.png) 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/63340c52-f9be-423a-a2f7-75f006844132) 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/154/builds/4538 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac-debug](https://ews-build.webkit.org/#/builders/165/builds/2142 "Passed tests") | [⏳ 🛠 wpe-cairo ](https://ews-build.webkit.org/#/builders/WPE-Cairo-Build-EWS "Waiting in queue, processing has not started yet") | | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/125626 "Built successfully and passed tests") | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/113673 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/135/builds/37848 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/143756 "Built successfully") | | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/132063 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/5724 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/38428 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/110545 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/5804 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/4910 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/110722 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/28063 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/4392 "Passed tests") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/115986 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/59473 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/5776 "Built successfully") | [  ~~🧪 mac-intel-wk2~~](https://ews-build.webkit.org/#/builders/137/builds/34301 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/165024 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/5623 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/69234 "Built successfully") | [❌ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/43122 "Found 5 new JSC stress test failures: microbenchmarks/emscripten-cube2hash-resizable.js.dfg-eager, stress/simple-regexp-test-folding.js.mini-mode, stress/to-lower-case-intrinsic-on-empty-rope.js.dfg-eager, stress/to-lower-case-intrinsic-on-empty-rope.js.no-llint, wasm.yaml/wasm/function-tests/memory-alignment.js.wasm-eager (failure)") | | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/163/builds/5867 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/5731 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

adetaylor commented on Source/WebCore/PAL/Configurations/PAL.xcconfig:
```diff
@@ -73,28 +67,4 @@ SWIFT_VERSION_XCODE_BEFORE_16 = 5.0;
-// Only for watchsimulator. Other platforms don't trigger Swift module builds for i386.
-EXCLUDED_ARCHS = $(EXCLUDED_ARCHS_$(WK_PLATFORM_NAME));
-EXCLUDED_ARCHS_watchsimulator = i386;
+OTHER_SWIFT_FLAGS = -Xcc -std=c++2b -I $(SRCROOT)/pal -I $(SRCROOT)/PAL/pal;
```
Please add `$(inherited)` (and please r+ https://github.com/WebKit/WebKit/pull/54527!) unless there's a specific reason you don't want to.

adetaylor replied:
Actually, scrub that - I've remembered we currently rely on the _absence_ of `$(inherited)` in this _specific_ case due to the reasons described in the commit message for https://commits.webkit.org/303537@main.

adetaylor commented on Source/WTF/wtf/RetainReleaseSwift.h:
```diff
@@ -25,16 +25,4 @@
 #pragma once
 
-#if HAVE(SWIFT_CPP_INTEROP) // FIXME: rdar://136787800
 #include <wtf/SwiftBridging.h>
```
Would it be straightforward to remove the `RetainReleaseSwift.h` header entirely, and switch all users to just including `SwiftBridging.h` instead? It doesn't seem to be used in the `Internal` repo so hopefully this is easy!

rr-codes replied:
yup! I am going to do that in a subsequent PR; I have like 6 or 7 more PRs to go 😔

mwyrzykowski reviewed (approved):
✅ based on @adetaylor's review

mwyrzykowski reviewed (commented):
Going to dismiss my approval while we discuss the safer-cpp bot

I was thinking it was a one time issue but sounds like not?

rr-codes committed:
[Swift in WebKit] Work towards modularizing PAL (part 1)
https://bugs.webkit.org/show_bug.cgi?id=303206
rdar://165493410

Reviewed by Mike Wyrzykowski.

Remove some settings from PAL and WebCore xcconfigs which are no longer used and/or no longer applicable

* Source/WTF/wtf/PlatformHave.h:
* Source/WebCore/Configurations/WebCore.xcconfig:
* Source/WebCore/PAL/Configurations/PAL.xcconfig:

Canonical link: https://commits.webkit.org/303763@main
